### PR TITLE
Added rsyslog to store HAProxy logs

### PR DESCRIPTION
**Pull request by marlan16**
Not a 100% tested and reviewed, but it should give you a preview of the solution.

HAProxy requires that syslog is running inside the container, so we need to use supervisord, rsyslog, cron and logrotate to make it work.

### CODE REVIEW

**Comment by david-richer-adsk** [comment]
Not possible to have rsyslog to send log to stdout and stderr?

**Comment by marlan16** [comment]
I saw something yesterday, I didn't tested it yet, but it was a symlink from /var/log/haproxy.log to /dev/stdout.

It might work, I will try it, it will remove the necessary to have logrotate.

**Comment by marlan16** [comment]
rsyslog is writing directly to /dev/stdout now, so we can see the logs directly from docker.

File: proxy/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+RUN apk add --no-cache \
+    socat \
+    supervisor \
+    rsyslog
+
+RUN sed -i '/imklog.so/d' /etc/rsyslog.conf
+
+COPY files_docker/etc/supervisord.conf /etc/supervisord.conf
+COPY files_docker/etc/rsyslog.d/haproxy.conf /etc/rsyslog.d/haproxy.conf
+COPY files_docker/usr/local/etc/haproxy.cfg /usr/local/etc/haproxy/haproxy.cfg
```
Comment (david-richer-adsk) [inline proxy/Dockerfile]: I would like to keep the haproxy.cfg and maintenance.http at the previous location

**Comment by david-richer-adsk** [comment]
Please add the log options to proxy too:

```
    logging:
      driver: "json-file"
      options:
        max-size: "2g"
        max-file: "20"
```

**Comment by marlan16** [comment]
Do we create a base image for the rsyslog modifications? Or we keep it that way with all modifications in this repo?

**Comment by david-richer-adsk** [comment]
We keep it in the repo, thanks for asking.

**Pull request merged**